### DISCUSSION TITLE
Maintenance cost and code line count

### DISCUSSION THREAD

**Discussion by pedia** [matrix-org/dendrite#3314] [General]
Noticed there are `108396` lines of Go in dendrite, Is it a huge amount of maintenance cost?

## code by module

|module | code |
|----|----|
|roomserver | 22156|
|syncapi | 19395|
|clientapi | 15627|
|userapi | 15542|
|federationapi | 13970|
|internal | 6243|
|setup | 4168|
|mediaapi | 2839|
|relayapi | 2117|
|appservice | 1214|

## code by folder name

|folder | code |
|---|---|
|storage |  **34937** |
|routing | 18032 |
|internal |  13992 |
|api |  2209 |
|producers/consumers | 4355 |
|util | 2441 |
|types |  2291 |

code line by [tokei](https://docs.rs/tokei/latest/tokei/), like:
- tokei clientapi/routing/ federationapi/routing/ mediaapi/routing/ relayapi/routing/ syncapi/routing/
- tokei clientapi/httputil/ clientapi/userutil/ internal/eventutil/ internal/httputil/ internal/sqlutil/ mediaapi/fileutils/ userapi/util/

## by storage type

|storage type | code |
|---|---|
| postgres | 11407 |
| sqlite3 | 12097 |
| tables/shared | 3807+5141 |

## Coverage of Unit test
```bash
go test ./... -cover -coverprofile=coverage.out
go tool cover -func=coverage.out
```
The total coverage is `23.3%`.

**Comment by S7evinK** [comment]
The problem with just `go test` is, that if e.g. Postgres is not set up beforehand, you're only testing the SQLite storage.
We're actually more in the area of having 50% unit test coverage. (see https://app.codecov.io/gh/matrix-org/dendrite/flags)

Combining everything (Unit Tests [run for Postgres and SQLite], instrumented tests when running Complement and Sytest [also in all combinations]), we currently have a code coverage of roughly 65%
